prisma/schema.prisma

```diff
@@ -1203,16 +1205,19 @@ model Worker {
   // the runs which retrieve the group keys
   groupKeyRuns GetGroupKeyRun[]
 
-  semaphore WorkerSemaphore?
+  slots WorkerSemaphoreSlot[]
 }
 
-model WorkerSemaphore {
-  // the parent worker
+model WorkerSemaphoreSlot {
```

Comment:
I think an index on the `workerId` would make sense here, because we're not writing semaphore slots very often (relative to the amount of step runs we're assigning), and we don't want the acquire query to slow down as we put more slots in the database.